frontend/public/src/containers/ProductDetailEnrollApp.js

```diff
@@ -149,10 +153,34 @@ export class ProductDetailEnrollApp extends React.Component<
     )
   }
 
+  updateDate(run: EnrollmentFlaggedCourseRun) {
+    let date = emptyOrNil(run.start_date) ? undefined : moment(new Date(run.start_date))
+    date = date ? date.utc() : date
+    const dateElem = document.getElementById('start_date')
+    if (dateElem) dateElem.innerHTML = `<strong>${formatPrettyDate(date)}</strong>`
+  }
+
   render() {
     const { courseRuns, isLoading, status } = this.props
     const csrfToken = getCookie("csrftoken")
-    let run = !this.getCurrentCourseRun() && courseRuns ? courseRuns[0] : this.getCurrentCourseRun()
+    let run = !this.getCurrentCourseRun() && courseRuns ? (
+      courseRuns[0]
+    ) : (
+      this.getCurrentCourseRun() && courseRuns ? (
```

Comment:
I'm having trouble understanding exactly what's being accomplished with the logic here.  The first if else is returning courseRuns[0] if `!this.getCurrentCourseRun() && courseRuns` is true.  The following else/if is then returning courseRuns[0] if `this.getCurrentCourseRun() && courseRuns` is false.

Reply:
The first condition is true when there is no `currentCourseRun` in current session then the `course run` will be selected from `courseRuns[0]` the second condition is true when there is course run is already available is session but no in `courseRuns` which is returned from the API it means the `currentCourseRun` from session is invalid and not for this course so it will return null. the `currentCourseRun` is stored into session so on page reload or in case of enrollment failure, the selected date (course run) will not be changed to default date (course run) but if you change the url to new course or open the new course in current tab (session) there will be previous course run in the session in that case this change is added so the previous course's `run` removed from the session and does not effect the new course

Reply:
This is helpful.  It might be good to add a small comment somewhere detailing the breakdown.